File: exercises/week02/alarmsystem-scala/src/ControlUnit.scala
import scala.collection.mutable

class ControlUnit(sensors: mutable.Buffer[Sensor]) {

  def pollSensors {
    //val sensors = new ListBuffer[Sensor]()
    //sensors += new FireSensor()
    //sensors += new SmokeSensor()
    for (sensor <- sensors) {
      if (sensor.isTriggered) {
        System.out.println("A " + sensor.getSensorType + " sensor was triggered at " + sensor.getLocation)
      }
      else {
        System.out.println("Polled " + sensor.getSensorType + " at " + sensor.getLocation + " successfully")
      }
    }
  }
}

object ControlUnit

File: exercises/week02/alarmsystem/src/ControlUnit.java
import java.util.ArrayList;
import java.util.List;

public class ControlUnit {

  public void pollSensors() {
    List<Sensor> sensors = new ArrayList<Sensor>();
    sensors.add(new FireSensor());
    sensors.add(new SmokeSensor());

    for (Sensor sensor : sensors) {
      if (sensor.isTriggered()) {
        System.out.println("A " + sensor.getSensorType() + " sensor was triggered at " + sensor.getLocation());
      } else {
        System.out.println("Polled " + sensor.getSensorType() + " at " + sensor.getLocation() + " successfully");
      }
    }
  }
}
